File: web/src/main/java/by/sorface/passport/web/config/security/jwt/KeyGeneratorUtils.java
package by.sorface.passport.web.config.security.jwt;

import lombok.experimental.UtilityClass;

import java.security.KeyPair;
import java.security.KeyPairGenerator;

@UtilityClass
public class KeyGeneratorUtils {

    public static KeyPair generateRsaKey(int keySize) {
        KeyPair keyPair;

        try {
            final var keyPairGenerator = KeyPairGenerator.getInstance("RSA");
            keyPairGenerator.initialize(keySize);

            keyPair = keyPairGenerator.generateKeyPair();
        } catch (Exception ex) {
            throw new IllegalStateException(ex);
        }

        return keyPair;
    }
}

File: web/src/main/kotlin/by/sorface/passport/web/config/security/jwt/KeyGeneratorUtils.kt
package by.sorface.passport.web.config.security.jwt

import java.security.KeyPair
import java.security.KeyPairGenerator

object KeyGeneratorUtils {

    fun generateRsaKey(keySize: Int): KeyPair {
        val keyPair: KeyPair

        try {
            val keyPairGenerator = KeyPairGenerator.getInstance("RSA")
            keyPairGenerator.initialize(keySize)

            keyPair = keyPairGenerator.generateKeyPair()
        } catch (ex: Exception) {
            throw IllegalStateException(ex)
        }

        return keyPair
    }

}
